# Module 1

## Question 1
Which of the following is not a suffix of CATATTAC?
* TATTAC
* TAC
* C
* CAT
```
Answer: CAT
```

## Question 2
What's the longest prefix of CACACTGCACAC that is also a suffix?
* CACACTG
* CACAC
* CAC
* C
```
Answer: CACAC
```

## Question 3
Which of the following is not a substring of GCTCAGCGGGGCA?
* GCT
* GCG
* GCA
* GCC
```
Answer: GCC
```

## Question 4
Starting around 2007, the cost of DNA sequencing started to decrease rapidly because more laboratories started to use:
* Second-generation sequencing
* DNA microarrays
* Double sequencing
* Sanger sequencing
```
Answer: Second-generation sequencing
```

## Question 5
Which of the following pieces of information is not included in a sequencing read in the FASTQ format:
* The sequence of base qualities corresponding to the bases
* A "name" for the read
* Which chromosome the read originated from
* The sequence of bases that make up the read
```
Answer: Which chromosome the read originated from
```

## Question 6
If read alignment is like "looking for a needle in a haystack," then the "haystack" is the:
* Sequencer
* Gene database
* Sequencing read
* Reference genome 
```
Answer: Reference genome
```

## Question 7
The Human Genome Project built the initial "draft" sequence of the human genome, starting from sequencing reads. The computational problem they had to solve was the:
* read alignment problem **XX**
* prime factorization problem
* gene finding problem
* de novo shutgun assembly problem
```
Answer: de novo shutgun assembly problem
```

## Question 8
If the length of the pattern is x and the length of the text is y, the minimum possible number of character comparisons performed by the naive exact matching algorithm is:
* x(y - x + 1)
* xy
* y - x + 1
* x + y
```
Answer: y - x + 1
``` 

## Question 9
If the length of the pattern is x and the length of the text is y, the maximum possible number of character comparisons performed by the naive exact matching algorithm is:
* x + y
* xy 
* y - x + 1
* x(y - x + 1)
```
Answer: x(y - x + 1)
```

## Question 10
Say we have a function that generates a random DNA string, i.e. the kind of string we would get by rolling a 4-sided die (A/C/G/T) over and over. We use the function to generate a random pattern P of length 20 and a random text T of length 100. Now we run the naive exact matching algorithm to find matches of P within T. We expect the total number of character comparisons we perform to be closer to the...
* maximum possible 
* minimum possible
```
Answer: minimum possible
```
