# Module 3 Quiz

## Question 1
How many lines make up each record contained in a FASTQ file?
* Three
* One
* Four
* Two
```
Answer: Four
```

## Question 2
Which of the following are components of the FASTQ format?
* Sequence identifier
* Quality score
* Sequence
* All of these options
```
Answer: All of these options
```

## Question 3
Which of the following is NOT a type of encoding of sequence quality scores in a FASTQ file?
* Ion torrent
* Sanger
* Solexa
* Illumina 1.3+
```
Answer: Ion torrent
```

## Question 4
Which of the following operations of quality control can be performed with Galaxyâ€™s NGS: QC and manipulation?
* FastQC: Read Quality reports
* Trim Sequences
* All of these options
* FASTA to FASTQ, and FASTQ to FASTA format conversion, Manipulate FASTQ
```
Answer: All of these options
```

## Question 5
Which of the following is NOT part of the FastQC report?
* Per sequence base quality
* Per sequence quality score
* Sequence Length distribution
* Per base sequence quality
```
Answer: Per sequence base quality
```

## Question 6
When do we filter and trim reads? Choose the correct option.
* Always filter reads. We want only high quality reads in our data.
* Always filter and trim reads, we want high quality equal length reads to gain flexibility of using any tool of choice during downstream analysis.
* Depends on tools used during downstream analysis.
* Always trim the reads. We want only equal length reads in our data.
```
Answer: Always filter and trim reads, we want high quality equal length reads to gain flexibility of using any tool of choice during downstream analysis.
```

## Question 7
What is ChIP sequencing most commonly used to measure?
* Ratios of miRNA to DNA
* Methylation of bases
* The locations of protein to DNA interaction
* Expression levels of particular proteins
```
Answer: The locations of protein to DNA interaction
```

## Question 8
What is MACS used for?
* Measure RNA levels
* Peak calling/reconstruction from ChIP-seq data
* Genome assembly using suffix trees and ChIP-seq data
* Detect sequence variants using both qPCR and ChIP-seq data
```
Answer: Peak calling/reconstruction from ChIP-seq data
```

## Question 9
What is the advantage of using a control in a ChIP sequencing experiment?
* Both options
* Allow MACS to calculate FDR.
* Determine background expectation of the number of peaks detected.
* Neither option

```
Answer: Both options
```
