## Question 1
How many alignments does the set contain?
```
Answer:
```

## Question 2
How many alignments show the read’s mate unmapped?
```
Answer:
```

## Question 3
How many alignments contain a deletion (D)?
```
Answer:
```

## Question 4
How many alignments show the read’s mate mapped to the same chromosome?
```
Answer:
```

## Question 5
How many alignments are spliced?
```
Answer:
```

## Question 6
How many alignments does the set contain?
```
Answer:
```

## Question 7
How many alignments show the read’s mate unmapped?
```
Answer:
```

## Question 8
How many alignments contain a deletion (D)?
```
Answer:
```

## Question 9
How many alignments show the read’s mate mapped to the same chromosome?
```
Answer:
```

## Question 10
How many alignments are spliced?
```
Answer:
```

Question 11
How many sequences are in the genome file?
```
Answer:
```

## Question 12
What is the length of the first sequence in the genome file?
```
Answer:
```

## Question 13
What alignment tool was used?
```
Answer:
```
## Question 14
What is the read identifier (name) for the first alignment?
```
Answer:
```

## Question 15
What is the start position of this read’s mate on the genome? Give this as ‘chrom:pos’ if the read was mapped, or ‘*” if unmapped.
```
Answer:
```

## Question 16
How many overlaps (each overlap is reported on one line) are reported?
```
Answer:
```

## Question 17
How many of these are 10 bases or longer?
```
Answer:
```

## Question 18
How many alignments overlap the annotations?
```
Answer:
```

## Question 19
Conversely, how many exons have reads mapped to them?
```
Answer:
```

## Question 20
If you were to convert the transcript annotations in the file “athal_wu_0_A_annot.gtf” into BED format, how many BED records would be generated?
```
Answer:
```
