# Module 3 Exam

## Question 1
How many sequences were in the genome?
```
Answer: 7
```

## Question 2
What was the name of the third sequence in the genome file? Give the name only, without the “>” sign.
```
Answer: Chr3
```

## Question 3
What was the name of the last sequence in the genome file? Give the name only, without the “>” sign.
```
Answer: mitochondria
```

## Question 4
How many index files did the operation create?
```
Answer: 6
```

## Question 5
What is the 3-character extension for the index files created?
```
Answer: bt2
```

## Question 6
How many reads were in the original fastq file?
```
Answer: 147354
```

## Question 7
How many matches (alignments) were reported for the original (full-match) setting? Exclude lines in the file containing unmapped reads.
```
Answer: 137719
```

## Question 8
How many matches (alignments) were reported with the local-match setting? Exclude lines in the file containing unmapped reads.
```
Answer: 141044
```

## Question 9
How many reads were mapped in the scenario in Question 7?
```
Answer: 137719
```

## Question 10
How many reads were mapped in the scenario in Question 8?
```
Answer: 141044
```

## Question 11
How many reads had multiple matches in the scenario in Question 7? You can find this in the bowtie2 summary; note that by default bowtie2 only reports the best match for each read.​
```
Answer: 43939
```

## Question 12
How many reads had multiple matches in the scenario in Question 8? Use the format above. You can find this in the bowtie2 summary; note that by default bowtie2 only reports the best match for each read.
```
Answer: 56105
```

## Question 13
How many alignments contained insertions and/or deletions, in the scenario in Question 7?
```
Answer: 2782
```

## Question 14
How many alignments contained insertions and/or deletions, in the scenario in Question 8?
```
Answer: 2614
```

## Question 15
How many entries were reported for Chr3?
```
Answer: 360295
```

## Question 16
How many entries have ‘A’ as the corresponding genome letter?
```
Answer: 1150985
```

## Question 17
How many entries have exactly 20 supporting reads (read depth)?
```
Answer: 1816
```

## Question 18
How many entries represent indels?
```
Answer: 1972
```

## Question 19
How many entries are reported for position 175672 on Chr1?
```
Answer: 2
```

## Question 20
How many variants are called on Chr3?
```
Answer: 398
```

## Question 21
How many variants represent an A->T SNP? If useful, you can use ‘grep –P’ to allow tabular spaces in the search term.
```
Answer: 392
```

## Question 22
How many entries are indels?
```
Answer: 320
``` 

## Question 23
How many entries have precisely 20 supporting reads (read depth)?
```
Answer: 2
```

## Question 24
What type of variant (i.e., SNP or INDEL) is called at position 11937923 on Chr3?
```
Answer: SNP
```
