# Understand the Problem

## Question 1
What is the type of tissue that was collected in this study?
* Liver
* Lymphoblastoid cell lines
* Dorsal Front Pre-frontal cortex (Brain)
* Hippocampus (Brain)
* Testes
* Kidney

### Answer
```
Dorsal Front Pre-frontal cortex (Brain)
```

## Question 2
How many samples did they sequence?
* 36 samples
* 6 samples
* 72 samples
* 16 samples
* 110 samples

## Question 3
How many of these samples were deposited on the Short Read Archive?
* 72 samples
* 16 samples
* 36 samples
* 6 samples
* 110 samples

## Question 4
How many age groups were included in the study?
* 1 age group
* 6 age groups
* 15 age groups
* 10 age groups
* 5 age groups

## Question 5
At what level did they summarize the data?
* Expressed Regions
* Genes
* Transcripts
* Exons

## Question 6
What differences were they looking for in their analysis
* Differences associated with Alzheimer’s disease
* Differences associated with age
* Differences between brain regions
* Differences associated with RNA quality (RIN)
* Differences between brain and kidney

## Question 7
Where can you get the raw data?
* At the authors’ website
* The Gene Expression Omnibus (GEO)
* The website of Nature Neuroscience
* The Short Read Archive (SRA)
* On Amazon S3

## Question 8
What do you expect the raw data to be? 
* FASTA files
* PDF files
* CSV files
* FASTQ files
* Image files

## Question 9
Which of the following were steps in the analysis pipeline they used for the RNA-seq data?
* Summarization into gene level counts based on known annotation
* Identification of differentially expressed regions
* Spliced alignment of reads to the genome
* Analysis of alternative splicing using exon-exon junctions
* Identification of differentially expressed genes based on known annotation
* Alignment of reads to known transcripts
* Identification of expressed regions using an analysis of coverage
* Alignment of reads to the genome
